lib/maths/CSignal.cc

```diff
@@ -182,12 +180,12 @@ void CSignal::autocorrelations(const TFloatMeanAccumulatorVec& values, TDoubleVe
     double mean = CBasicStatistics::mean(moments);
     double variance = CBasicStatistics::maximumLikelihoodVariance(moments);
 
-    TComplexVec f;
-    f.reserve(n);
+    TComplexVec f(n, TComplex{});
```

Comment:
The loop below contains `push_back` and `emplace_back` calls on this vector. Is the final size meant to be `n`? If so I think it will end up too big unless the loop contents are changed too.

Having said that, given that a couple of branches inside the loop set values to (0, 0) you’re probably right that it’s clearest to just initialize a correctly sized vector to all (0, 0) and then change the elements that need changing.

Reply:
Yeah, I've stepped through this several times. The call to `emplace_back` happens after the vector has been truncated by `resize`.

Actually, on the face of it I think the code was probably fine as it stood but the change keeps `Valgrind` happy. The other option then is to leave this bit of the code alone and to simply add the offending stack trace to a `Valgrind` suppression file (that would need to be created and committed as part of this PR).

Which leads me to another suggestion. Should we include `Valgrind` checks as part of the Jenkins CI builds for PRs? Maybe we could also run a full `Valgrind` check once a week or so, over the weekend?